webservices/phoenix_deployment.md

```diff
@@ -0,0 +1,192 @@
+
+Note: these steps could all be automated using some sort of provisioning tool (ie elastic beanstalk, terraform, chef, puppet, etc) but is outside the scope of this guide.
+
+## Setup app for release
+
+[Distillery](https://github.com/bitwalker/distillery) is the package we'll be using to actually generate the releases we'll be deploying to production. If using phoenix, follow the installation and setup directions starting [here](https://hexdocs.pm/distillery/use-with-phoenix.html). If not using phoenix, begin [here](https://hexdocs.pm/distillery/walkthrough.html). Once distillery is installed and you are able to generate a release via the command `mix release` we're ready to move on to edeliver, which will make the process of building and deploying apps much easier than needing to manually copy and unpack the tarballs generated by distillery.
+
+Begin [here](https://github.com/edeliver/edeliver#quick-start) and walk through the steps to get edeliver setup. When you reach the `.deliver/config` portion make sure to substitute in values that match with the instances that you've created.
+
+```bash
+# .deliver/config
+
+APP="app_name"
+
+BUILD_HOST="ip_address_to_build_instance"
+BUILD_USER="ubuntu"
+BUILD_AT="/tmp/edeliver/app_name/builds"
+
+PRODUCTION_HOSTS="ip_address_to_prod_instance"
+PRODUCTION_USER="ubuntu"
+DELIVER_TO="/home/ubuntu/app"
+```
+
+Note: If your app requires additional build steps, such as `mix phx.digest` or `npm install` see the article [here](https://github.com/edeliver/edeliver/wiki/Run-additional-build-tasks) on how to setup those hooks to run on build/deployment.
+
+## Migrations
+
+If your app requires a database (most will), see [here](https://hexdocs.pm/distillery/running-migrations.html) on how to set up a migration module that you can run either manually or configure using edelivery to run automatically on deploy.
+
+## Configuration
+
+One of the main sticking points for a successful deployment is figuring out where and how to handle configuration values. You don't want to commit and store the values, so you need to maintain them either using elixir's `prod.secret.exs` convention or in the environment of your server. This leads to another issue with configuration. Since elixir applications are compiled, the values need to be present at compile time to be included in the app. So just haven't the env vars setup in production will work for values that are fetched at run time, but not compile time.
+
+There are two ways of handling this.
+
+### Compile time
+
+For compile time env vars, you need to ensure the values you need are loaded into the build server's environment.
+
+```bash
+# ~/.profile
+
+..
+export CONFIG_VALUE=foo
+..
+```
+note: `~/.profile` is automatically loaded into the environment during edeliver operations, so it's a good place to store environment variables on the build machine.
```

Comment:
What about having `.env` should live at the app root, and in `~/.profile` you just have `source ~/app/blahblahapp/.env`